enhancements/ota/distribute-secondary-metadata-as-container-image.md

```diff
@@ -0,0 +1,165 @@
+This image will be available for public consumption, and will be used by the publicly available Cincinnati instances operated by Red Hat.
+The image can be transferred and served to Cincinnati instances operated by any party, including but not limited to network-restricted environments.
+The source of the container image, namely the cincinnati-graph-data, will remain publicly available and its workflows unaffected by this enhancement.
+
+## Motivation
+
+### Goals
+- The container image will become the primary way of consuming and distributing the content of the cincinnati-graph-data repository.
+- As long as the primary release metadata will be packaged, distributed and consumed in form of container images as well, this streamlines all transport mechanisms required for data consumption by Cincinnati, as well as mirroring and serving its consumed content.
+- Every change to the master branch of the cincinnati-graph-data results in a new container image containing the unmodified working tree files of the new revision.
+- The image will be pushed to a publicly available container image registry und a well-known name.
+
+### Non-Goals
+- Changes to the GitHub repository workflow or data format and make progress.
+- Implementation details about the streamlining of the mirroring process for primary and secondary release metadata.
+
+## Proposal
+
+### User Stories [optional]
+
+#### Content mirroring for Cincinnati deployments
+A user might want to mirror all content which is required by Cincinnati.
+
+1. The user sets up a container image registry in the target environment.
+2. Optionally, if the target environment does not have internet access, the user downloads the release container images and the desired graph-data container image to a transfer medium.
+3. The user pushes the release and graph-data images to the target container image registry
+4. The user can now configure the Cincinnati instance in the target environment to consume the content from the available container image registry
+
+### Implementation Details/Notes/Constraints [optional]
+* CPaaS needs to be configured
+
+### Risks and Mitigations
+
+### Container image content integrity
+The content of the cincinnati-graph-data repository may be altered by the process of being packaged by CPaaS and distributed via the container image registry.
+
+#### Mitigation
+A mitigation strategy is to sign the resulting image in the CPaaS pipeline, publish the signatures, and have Cincinnati verify the image after download.
+
+### cincinnati-graph-data source repository integrity
+Not strictly related to this enhancement, is the fact that we are not signing the commits in the source repository in the first place.
+By this we effectively trust GitHub not to change the content.
+
+#### Mitigation
+Enforce signed commits in the source repository.
```

Comment:
This has been raised to OpenShift architects and explicitly rejected before we moved to the graph-data repo workflow.  But sure, it's a risk.  I'd suggest explaining the current position with something like:

> OpenShift architects are aware of this risk and currently do not consider it sufficiently serious to want mitigation.  If we decide to mitigate in the future, having build tooling require commits to be signed by a hard-coded list of authorized maintainers would be one possible approach.

Reply:
Pasted in verbatim, thanks ;-)